File: 53-maximum-subarray/53-maximum-subarray.go
func maxSubArray(nums []int) int {
    maxSoFar := -1 << 31
    currentMax := 0
    for i := range nums {
        currentMax += nums[i]
        if currentMax > maxSoFar {
            maxSoFar = currentMax
        }
        if currentMax < 0 {
            currentMax = 0
        }
    }
    return maxSoFar
}

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maxSum = nums[0];
        for(int i : nums)
        {
            if(sum + i > 0)
                sum = sum + i;
            else
                sum = 0;
            if(sum > maxSum && sum!= 0)
                maxSum = sum;
            if(i > maxSum)
                    maxSum = i;
        }
        return maxSum;
    }
};